rpcd/playbooks/ironic-pre-config.yml

```diff
@@ -0,0 +1,125 @@
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ cidr_networks_ironic }}"
+        config_type: "yaml"
+
+    - name: Add IPs to used_ips list
+  
+    - name: Setup global overrides
+      config_template:
+        src: "/etc/openstack_deploy/openstack_user_config.yml"
+        dest: "/etc/openstack_deploy/openstack_user_config.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_networks }}"
+        config_type: "yaml"
+
+    - name: Define group hosts
+      config_template:
+        src: "/etc/openstack_deploy/openstack_user_config.yml"
+        dest: "/etc/openstack_deploy/openstack_user_config.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ group_hosts_ironic }}"
+        config_type: "yaml"
+
+    - name: Setup endpoints for swift/ironic health checks
+       config_template:
+        src: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        dest: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_endpoint_config }}"
+        config_type: "yaml"
+      
+    - name: Configure neutron DHCP to be non-authorative
+      config_template:
+        src: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        dest: "/etc/openstack_deploy/user_osa_variables_overrides.yml"
+        owner: "root"
+        group: "root"
+        mode: "0644"
+        config_overrides: "{{ ironic_neutron_dhcp_config }}"
+        config_type: "yaml"
+
+    - name: Setup neutron network
+      shell: "neutron net-create --shared --provider:physical_network tftp --provider:network_type flat tftp"
+
+    - name: Setup neutron subnet
+      shell: "neutron subnet-create --name ironic-tftp --allocation-pool start=<NON_ALLOCATED_TFTP_RANGE_START>,end=<NON_ALLOCATED_TFTP_RANGE_END> --dns-nameserver=4.4.4.4 tftp <TFTP_NETWORK>/<TFTP_NETMASK>"
+
+  vars:
+    cidr_networks_ironic: 
+      cidr_networks:
+           tftp: {{ tftp_network }}{{ tftp_netmask }}
+           ironic-ipmi: {{ ironic-ipmi_network }}{{ ironic-ipmi_netmask }}
+    ironic_networks:
+      - network:
+          container_bridge: "br-ironic-ipmi"
+          container_type: "veth"
+          container_interface: "eth_ipmi"
+          ip_from_q: "ironic-ipmi"
+          type: "raw"
+          group_binds:
+          - ironic-infra_hosts
+      - network:
+          container_bridge: "br-tftp"
+          container_type: "veth"
+          container_interface: "eth_tftp"
+          ip_from_q: "ironic-ipmi"
+          type: "flat"
+          net_name: "tftp"
+          ip_from_q: "tftp"
+          group_binds:
+          - neutron_linuxbridge_agent
+          - ironic_all
+    group_hosts_ironic:
+      ironic-compute_hosts:
+        infra01:
+          ip: {{ infra01_mgmt_ip }}
```

Comment:
These need quotations around them, though we also should figure out how to make this work for both AIO and multi-node AIO builds in the gate. Maybe we can dynamically set the values of ironic-compute_hosts and ironic-infra_hosts based on the existing inventory for infra nodes?